### PR TITLE
instance-based adv loss

### PR DESCRIPTION
**Pull request by bplank**
Hi @sebastianruder 
changes to instance-based adv loss.
Barbara

### CODE REVIEW

File: src/amt3.py [hunk 1]
```diff
@@ -469,13 +473,12 @@ def predict(self, word_indices, char_indices, task_id, train=False,
 
                 if domain_id is not None:
                     # flip the gradient when back-propagating through here
-                    adv_input = [dynet.flip_gradient(adv_in) for adv_in in concat_layer]
+                    adv_input = [dynet.flip_gradient(concat_layer[-1])] # last state
```
Comment (sebastianruder) [inline src/amt3.py]: We don't need to use a list here anymore as it's only one element, right? Then we also don't need to average below.

**Comment by bplank** [comment]
good catch yes! let me fix this

**Comment by bplank** [comment]
Ok reverted it.

**Comment by sebastianruder** [comment]
Looks good and can be merged.

**Pull request merged**